### PR TITLE
Include large API keys in the JSON body instead of as a header

### PR DESCRIPTION
**Pull request by kylegibson**
This mirrors the behavior of the JS client. The algolia API server will reject requests with headers that are too long.

### CODE REVIEW

**Comment by kylegibson** [comment]
The reason why travis tests are failing is because travis disables the global secure environment data for remote/untrusted PRs. Took me a while to get tox running locally to discover that this was what was happening.

File: tests/test_index.py [hunk 1]
```diff
@@ -277,6 +279,33 @@ def test_search(self):
         res_ids = [elt['objectID'] for elt in res['hits']]
         self.assertIn(self.objectIDs[2], res_ids)
 
+    def test_search_with_short_secured_api_key(self):
+        old_key = self.client.api_key
+
+        secured_api_key = self.client.generate_secured_api_key(
+            os.environ['ALGOLIA_API_KEY_SEARCH'],
```
Comment (kylegibson) [inline tests/test_index.py]: I discovered while running tox locally that ``ALGOLIA_API_KEY`` needed to be set to the admin key. However, when generating a secured api key, the admin key doesn't work. So, for these two new test cases, we need an additional environment variable for the search api key. That needs to be configured in `.travis.yml` as a secured parameter.

File: tests/test_old.py [hunk 2]
```diff
@@ -59,11 +59,11 @@ def test_wrong_app_id(self):
...
-      try:
-          client = algoliasearch.Client(os.environ['ALGOLIA_APPLICATION_ID'], os.environ['ALGOLIA_API_KEY'], ["fakeapp-1.algolianet.com", "fakeapp-2.algolianet.com", os.environ['ALGOLIA_APPLICATION_ID'] + ".algolianet.com"])
-          client.listIndexes
-      except algoliasearch.AlgoliaException as e:
-          self.assertTrue(false)
+        try:
+            client = algoliasearch.Client(os.environ['ALGOLIA_APPLICATION_ID'], os.environ['ALGOLIA_API_KEY'], ["fakeapp-1.algolianet.com", "fakeapp-2.algolianet.com", os.environ['ALGOLIA_APPLICATION_ID'] + ".algolianet.com"])
+            client.listIndexes
+        except algoliasearch.AlgoliaException as e:
+            self.assertTrue(False)
```
Comment (kylegibson) [inline tests/test_old.py]: This change is unrelated to the API key issue. Here I am just fixing the indentation and a typo (``false`` => ``False``).

File: tests/test_index.py [hunk 3]
```diff
@@ -277,6 +279,33 @@ def test_search(self):
...
+
+        tags = set('x{}'.format(100000 + i) for i in range(1000))
+        secured_api_key = self.client.generate_secured_api_key(
+            os.environ['ALGOLIA_API_KEY_SEARCH'],
+            dict(filters=' OR '.join(tags)),
+        )
+        assert len(secured_api_key) > MAX_API_KEY_LENGTH
+        self.client.api_key = secured_api_key
+        res = self.index.search('')
+        self.assertEqual(res['nbHits'], 0)
```
Comment (kylegibson) [inline tests/test_index.py]: Since none of the initially added records have any of the tags, the result is 0. I could add a record in the test that has at least one of the tags, but it might not really be necessary. The key here is that no exception occurs.

**Comment by PLNech** [comment]
Hi @kylegibson, thank you for this contribution!

I'm closing this PR and reopening it as #84 to use the Environment Variables in Travis 😉

**Comment by PLNech** [comment]
Published in [1.9.2](https://github.com/algolia/algoliasearch-client-python/releases/tag/1.9.2).
